zerver/openapi/zulip.yaml

```diff
@@ -9124,6 +9124,19 @@ paths:
                     server sent user presence data for all users who had been active in the
                     last two weeks unconditionally.
                   example: 5
+                history_limit_days:
+                  type: integer
+                  description: |
+                    Limits how far back in time to fetch user presence data. If not specified,
+                    defaults to 14 days. A value of N means that the oldest presence data
+                    fetched will be from at most N days ago.
+
+                    Note that this is only useful during the initial user presence data fetch,
+                    as subsequent fetches should use the `last_update_id` parameter, which
+                    will act as the main limit on how much presence data is returned.
```

Comment:
I think this is suggesting that both this new `history_limit_days` parameter and the `last_update_id` will apply as limits: we'll return only those entries that both are after `last_update_id` and are in the last `history_limit_days` days.

That behavior sounds good. Let's describe the interaction explicitly, though.

Reply:
I think the currently implemented interaction is that `history_limit_days` is ignored if `last_update_id` is present and has a real value; so we should definitely document that if we're doing it that way. I certainly think few clients would want to combine those operators; if you're doing a cursor, it's quite unlikely that combining the cursor with a time limit is going to do anything other than make the database query less efficient.

In the rare case that a client has a `last_update_id` that is older than the `history_limit_days` that it wants, probably the correct behavior for such a client would be to just discard its existing presence data (which is all stale) and not pass `last_update_id`, rather than combining these. So I think what we have implemented is fine, but we should document this detail.

Reply:
Hmm, no - the current implementation doesn't make a special case for this combination of parameters. It'll handle them both in the expected way, by including both in its filter expression. That's why this doc notes that combining them is not really useful, even if accepted.

Reply:
Hmm oh yeah, I was confused here - that was the original behavior I added here, but then we changed that adding this block:
```
    if last_update_id_fetched_by_client is None or last_update_id_fetched_by_client <= 0:
        # If the client already has fetched some presence data, as indicated by
        # last_update_id_fetched_by_client, then filtering by last_connected_time
        # is redundant, as it shouldn't affect the results.
        kwargs["last_connected_time__gte"] = fetch_since_datetime
````

Tweaked the doc now:
```
                    Note that this is only useful during the initial user presence data fetch,
                    as subsequent fetches should use the `last_update_id` parameter, which
                    will act as the limit on how much presence data is returned. `history_limit_days`
                    is ignored if `last_update_id` is passed with a value greater than `0`,
                    indicating that the client already has some presence data.

```